creator-node/src/middlewares.js

```diff
@@ -98,6 +100,41 @@ async function ensurePrimaryMiddleware (req, res, next) {
   next()
 }
 
+/** Blocks writes if node has used over `maxStorageUsedPercent` of its capacity. */
+async function ensureStorageMiddleware (req, res, next) {
+  // Get storage data and max storage percentage allowed
+  const [storagePathSize, storagePathUsed] = await getMonitors([
+    MONITORS.STORAGE_PATH_SIZE,
+    MONITORS.STORAGE_PATH_USED
+  ])
+
+  const maxStorageUsedPercent = config.get('maxStorageUsedPercent')
+
+  // Check to see if CNode has enough storage
+  let hasEnoughStorage = hasEnoughStorageSpace({
+    storagePathSize,
+    storagePathUsed,
+    maxStorageUsedPercent
+  })
+
+  if (hasEnoughStorage) {
+    next()
+  } else {
+    const errorMsg = `Node is reaching storage space capacity. Current usage=${(100 * storagePathUsed / storagePathSize).toFixed(2)}% | Max usage=${maxStorageUsedPercent}%`
+    req.logger.error(errorMsg)
+    return sendResponse(
+      req,
+      res,
+      errorResponseBadRequest(
+        {
+          msg: errorMsg,
+          state: 'NODE_REACHED_CAPACITY'
```

Comment:
setting this state in error msg looks like a new pattern. whats benefit?

Reply:
I wanted to ensure that the error received in the unit tests are from reaching capacity, and not another error. I didn't want to check if the error message contains a substring of the word "capacity reached" as error messages are prone to change. although this is a new pattern, I thought adding a state field in the error message would allow us to explicitly make sure that this the node capacity error is thrown

Reply:
sounds good! interesting approach to evaluate longer term 👍